Sources/web3swift/Operations/WriteOperation.swift

```diff
@@ -32,4 +32,25 @@ public class WriteOperation: ReadOperation {
         // MARK: Sending Data flow
         return try await web3.eth.send(transaction)
     }
+
+    public func depploy(password: String, policies: Policies = .auto) async throws -> TransactionSendingResult {
+   //        try await transaction.resolve(provider: web3.provider)
+           try await policyResolver.resolveAll(for: &transaction, with: policies)
+
+           guard let attachedKeystoreManager = self.web3.provider.attachedKeystoreManager else {
+               throw Web3Error.inputError(desc: "Failed to locally sign a transaction")
+           }
+
+           do {
+               let account = transaction.from ?? transaction.sender ?? EthereumAddress.contractDeploymentAddress()
+               var privateKey = try attachedKeystoreManager.UNSAFE_getPrivateKeyData(password: password, account: account)
+               defer { Data.zero(&privateKey) }
+               try transaction.sign(privateKey: privateKey, useExtraEntropy: false)
+           } catch {
+               throw Web3Error.inputError(desc: "Failed to locally sign a transaction")
+           }
+
+           guard let transactionData = transaction.encode(for: .transaction) else { throw Web3Error.dataError }
+           return try await web3.eth.send(raw: transactionData)
+       }
```

Comment:
Was under the impression there is already a `deploy` function but can't remember where? @yaroslavyaroslav @JeneaVranceanu

Reply:
[Here it is](https://github.com/skywinder/web3swift/blob/develop/Tests/web3swiftTests/localTests/AdvancedABIv2Tests.swift#L15-L28). I'd also doubt about is this method really necessary?

Reply:
that one does not end up sending eth transaction raw so the new one is needed

Reply:
What about to change `writeToChain` one to send raw tx in both cases rather then add another method?

Reply:
to confirm, I should update the writeToChain with my deploy adjustments and then delete deploy?

Reply:
Actually yes.

Reply:
will do :)

Reply:
I support the removal of the `EthereumContract.deploy` and `Contract.prepareDeploy` functions. These both are just a convenience functions and in my opinion they never belonged to `Contract` and `EthereumContract`.
I suppose what was their initial intention is to add a way to create a deployment transaction object that will deploy a smart contract with ABI you used to initialize `EthereumContract`.

On that note, I'd also strongly advise creating an example that we can please in the docs that would be roughly named "How to deploy a smart contract?". 
It should contain the bare minimum example + an example where a smart contract has a constructor and how do we set the argument to that constructor.

Reply:
It looks like I've gone the wrong path while reviewing this PR 🙂
Anyway, the comment is valid and can be addressed in a different PR.

Reply:
Combining both `writeToChain` and this `delpoy` function you've added here is what I suggest.
`sendRaw` - if `false` doesn't attempt signing. Set to `true` by default.
+ a few minor tweaks to the error messages.

```swift
public func writeToChain(password: String,
                         policies: Policies = .auto,
                         sendRaw: Bool = true) async throws -> TransactionSendingResult {
    try await policyResolver.resolveAll(for: &transaction, with: policies)

    guard sendRaw else {
        return try await web3.eth.send(transaction)
    }

    guard let attachedKeystoreManager = web3.provider.attachedKeystoreManager else {
        throw Web3Error.inputError(desc: "Failed to locally sign a transaction. Web3 provider doesn't have keystore attached.")
    }

    do {
        try Web3Signer.signTX(transaction: &transaction,
                              keystore: attachedKeystoreManager,
                              account: transaction.from ?? transaction.sender ?? EthereumAddress.contractDeploymentAddress(),
                              password: password)
    } catch {
        throw Web3Error.inputError(desc: "Failed to locally sign a transaction. \(error.localizedDescription)")
    }
    guard let transactionData = transaction.encode(for: .transaction) else { throw Web3Error.dataError }
    return try await web3.eth.send(raw: transactionData)
}
```

Reply:
oh, that one is nice I'm up to @JeneaVranceanu suggestion about how to implement that. @pharms-eth